crossdock/server/oneway/echo.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package oneway
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"go.uber.org/yarpc"
+	"go.uber.org/yarpc/encoding/raw"
+	"go.uber.org/yarpc/transport"
+	"go.uber.org/yarpc/transport/http"
+)
+
+const callBackAddr = "http://127.0.0.1:8089"
+
+// EchoRaw implements the echo/raw procedure.
+func EchoRaw(ctx context.Context, reqMeta yarpc.ReqMeta, body []byte) error {
+	//make call back to the client
+	callHome(body)
+	return nil
+}
+
+// EchoJSON implements the echo/json procedure.
+func EchoJSON(ctx context.Context, reqMeta yarpc.ReqMeta, body map[string]interface{}) error {
+	token := body["token"].(string)
```

Comment:
This function can accept a `struct { Token string }` instead. It can be a separate type or an inline anonymous struct type.